### PR TITLE
Make contractToRead shared code [1/n]

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Initial work to converge the v1 and v2 contract readers (as they share a lot of code).
This also begins the work of create a hook specifically for the contract read value, and allowing the ability to update the read value in a simple way (coming later in stack).

this code can be shared amongst these two files

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1869** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1869" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1869?utm_source=stack-comment).

File: src/utils/contractToRead.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Contract } from '@ethersproject/contracts'
+
+type ContractConfig<T> = T | Contract | undefined
+
+export function contractToRead<T extends string>(
```
Comment (peripheralist) [inline src/utils/contractToRead.ts:5]: since this util is outside the scope of actually reading the contract, maybe name it something like "getContract"?

Comment (wraeth-eth) [reply src/utils/contractToRead.ts]: Solid - totally agree. `getContract` is a much better name

**Review by peripheralist — APPROVED** [review]
